core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -361,7 +381,9 @@ public void masterRegister(long masterId, RegisterMasterTOptions options)
 
     List<ConfigProperty> configList = options.getConfigList().stream()
         .map(ConfigProperty::fromThrift).collect(Collectors.toList());
-    mMasterConfigChecker.registerNewConf(masterId, configList);
+    MasterAddress masterAddress = master.getMasterAddress();
+    mMasterConfigRecord.registerNewConf(new Address(masterAddress.getHost(),
```

Comment:
Do we need to regenerate the conf report here?

Right now there is a potential source of bugs if we modify `mMasterConfigRecord` without calling `regenerateReport()`. It would be good to have a general-purpose solution so that all changes to `mMasterConfigRecord` will automatically cause the config report to be regenerated. There are a few ways we could do this:
1. Create a `ConfigurationManager` class that wraps around `mMasterConfigRecord` and `mConfigChecker`, providing methods like `registerNewConf`/`lostNodeFound`. When those methods are called, the `ConfigurationManager` can call the corresponding method on its `mMasterConfigRecord`, and also call `mConfigChecker.regenerateReport()` to guarantee that the report is always up to date.
2. Add functionality for adding listeners to `ServerConfigurationRecord` so that `mConfigChecker` can listen for changes to `mMasterConfigRecord` and regenerate the report whenever a change occurs.